src/v/cluster/tm_stm.h

```diff
@@ -213,6 +213,10 @@ class tm_stm final : public persisted_stm {
 
     absl::btree_set<kafka::transactional_id> get_expired_txs();
 
+    using ntp_filter = ss::noncopyable_function<bool(const model::ntp&)>;
+    absl::btree_set<kafka::transactional_id>
+    get_txs_matching_ntp_filter(ntp_filter&&);
```

Comment:
nit: this can be a template function leveraging concepts instead of virtual dispatch